### PR TITLE
Remove the `source.request.syntacticrename` request

### PR DESCRIPTION
**Pull request by ahoppen**
`source.request.syntacticrename` isn’t used anywhere and `source.request.find-syntactic-rename-ranges` should be used instead.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: tools/SourceKit/include/SourceKit/Core/LangSupport.h [hunk 1]
```diff
@@ -1189,16 +1189,11 @@ class LangSupport {
...
-                               ArrayRef<RenameLocations> RenameLocations,
-                               ArrayRef<const char*> Args,
-                               CategorizedEditsReceiver Receiver) = 0;
-
   virtual void findRenameRanges(llvm::MemoryBuffer *InputBuf,
                                 ArrayRef<RenameLocations> RenameLocations,
                                 ArrayRef<const char *> Args,
                                 CategorizedRenameRangesReceiver Receiver) = 0;
-  virtual void
+   virtual void
```
Comment (bnbarham) [inline tools/SourceKit/include/SourceKit/Core/LangSupport.h:1196]: Accidental? Does clang-format not remove this?

**Review by bnbarham — APPROVED** [review]

**Review by hamishknight — APPROVED** [review]

File: tools/SourceKit/include/SourceKit/Core/LangSupport.h (hunk 1 above)
Comment (ahoppen) [reply tools/SourceKit/include/SourceKit/Core/LangSupport.h]: It does. I just didn’t push after running `git-clang-format` 🙈

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**